Tools/Scripts/libraries/webkitapipy/webkitapipy/sdkdb_from_build_products.py

```diff
@@ -0,0 +1,83 @@
+import argparse
+import os
+import shutil
+import stat
+import subprocess
+import tempfile
+from typing import cast
+from pathlib import Path
+
+
+def is_executable_file(path):
+    ext = os.path.splitext(path)[1]
+    mode = stat.S_IFREG | stat.S_IXUSR
+    return ext == '.a' or ext in ('', '.dylib') and \
+        os.stat(path).st_mode & mode == mode
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='wrapper around `tapi sdkdb` to produce a partial sdkdb '
+        'from Mach-O build products.')
+    parser.add_argument('products', nargs='+', type=Path,
+                        help='frameworks, dylibs, or root directories to '
+                        'include in generated sdkdb')
+    parser.add_argument('--sdk-dir',
+                        help='SDK to use when parsing public headers')
```

Comment:
Might be good to say that it defaults to SDKROOT if available. But what if neither is available? From a cursory glance, looks like the script will misbehave.